pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/cockroach-go-testserver-upgrade-to-master_test: TestLogic_mixed_version_upgrade_repair_descriptors failed

pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/cockroach-go-testserver-upgrade-to-master_test.TestLogic_mixed_version_upgrade_repair_descriptors [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12731079?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12731079?buildTab=artifacts#/) on master @ [7f00287df9a1a39b410293ef03ddffb550d6d3f9](https://github.com/cockroachdb/cockroach/commits/7f00287df9a1a39b410293ef03ddffb550d6d3f9):


```
=== RUN   TestLogic_mixed_version_upgrade_repair_descriptors
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/5cf022a3d4669c7cd3127fd7792ab475/logTestLogic_mixed_version_upgrade_repair_descriptors3418150709
    test_log_scope.go:81: use -show-logs to present logs inline
    logic.go:2908: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7883/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/cockroach-go-testserver-upgrade-to-master_test_/cockroach-go-testserver-upgrade-to-master_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_repair_descriptors:10: SELECT version LIKE '%23.1-%' FROM [SHOW CLUSTER SETTING version]
        expected:
            true
            
        but found (query options: "retry") :
            false
            
[17:19:16] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7883/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/cockroach-go-testserver-upgrade-to-master_test_/cockroach-go-testserver-upgrade-to-master_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_repair_descriptors: 2 statements
[17:19:16] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7883/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/cockroach-go-testserver-upgrade-to-master_test_/cockroach-go-testserver-upgrade-to-master_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_repair_descriptors with config cockroach-go-testserver-upgrade-to-master: 2 tests, 1 failures
    logic.go:3998: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7883/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/cockroach-go-testserver-upgrade-to-master_test_/cockroach-go-testserver-upgrade-to-master_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_repair_descriptors:15: error while processing
    logic.go:3998: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7883/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/cockroach-go-testserver-upgrade-to-master_test_/cockroach-go-testserver-upgrade-to-master_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_repair_descriptors:15: too many errors encountered, skipping the rest of the input
    panic.go:523: -- test log scope end --
cockroach logs captured in: /artifacts/tmp/_tmp/5cf022a3d4669c7cd3127fd7792ab475/cockroach-logs3257686233
test logs left over in: /artifacts/tmp/_tmp/5cf022a3d4669c7cd3127fd7792ab475/logTestLogic_mixed_version_upgrade_repair_descriptors3418150709
--- FAIL: TestLogic_mixed_version_upgrade_repair_descriptors (136.07s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_mixed_version_upgrade_repair_descriptors.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-33613

--- a/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl
+++ b/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl
@@ -1,5 +1,13 @@
 # LogicTest: cockroach-go-testserver-upgrade-to-master
 
+# This test verifies that when a cluster is upgraded, it preserves the TTL
+# job configuration on a table.
+
+query T
+SELECT version FROM [SHOW CLUSTER SETTING version]
+----
+23.1
+
 statement ok
 CREATE TABLE tbl (
   id INT PRIMARY KEY
@@ -7,8 +15,10 @@ CREATE TABLE tbl (
 
 upgrade all
 
+# Verify that the cluster version upgrades have begun. Note that the first
+# cluster upgrade is the one that repairs all descriptors.
 query B retry
-SELECT version LIKE '%23.1-%' FROM [SHOW CLUSTER SETTING version]
+SELECT version LIKE '%23.2-%' FROM [SHOW CLUSTER SETTING version]
 ----
 true
 

--- a/pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/generated_test.go
+++ b/pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/generated_test.go
@@ -176,11 +176,11 @@ func TestLogic_mixed_version_udf_mutations(
 	runLogicTest(t, "mixed_version_udf_mutations")
 }
 
-func TestLogic_mixed_version_upgrade_repair_descriptors(
+func TestLogic_mixed_version_upgrade_preserve_ttl(
 	t *testing.T,
 ) {
 	defer leaktest.AfterTest(t)()
-	runLogicTest(t, "mixed_version_upgrade_repair_descriptors")
+	runLogicTest(t, "mixed_version_upgrade_preserve_ttl")
 }
 
 func TestLogic_pg_lsn_mixed(

--- a/pkg/sql/schemachanger/comparator_generated_test.go
+++ b/pkg/sql/schemachanger/comparator_generated_test.go
@@ -1083,9 +1083,9 @@ func TestSchemaChangeComparator_mixed_version_udf_mutations(t *testing.T) {
 	var logicTestFile = "pkg/sql/logictest/testdata/logic_test/mixed_version_udf_mutations"
 	runSchemaChangeComparatorTest(t, logicTestFile)
 }
-func TestSchemaChangeComparator_mixed_version_upgrade_repair_descriptors(t *testing.T) {
+func TestSchemaChangeComparator_mixed_version_upgrade_preserve_ttl(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	var logicTestFile = "pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_repair_descriptors"
+	var logicTestFile = "pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl"
 	runSchemaChangeComparatorTest(t, logicTestFile)
 }
 func TestSchemaChangeComparator_multi_region(t *testing.T) {

